src/operator.rs

```diff
@@ -600,7 +612,16 @@ impl UnaryPreAggFunction for Parse {
     fn process(&self, rec: Record) -> Result<Option<Record>, EvalError> {
         let matches = self.matches(&rec)?;
         match matches {
-            None => Ok(None),
+            None => match self.options.drop_nonmatching {
+                true => Ok(None),
+                false => {
+                    let rec = self
+                        .fields
+                        .iter()
+                        .fold(rec, |rec, field| rec.put(field, data::Value::None));
```

Comment:
Maybe make 'rec' mutable and use for_each() ?  Not sure what the style is...

Reply:
this is OK as is. This is probably a little more readable though: (as @tstack mentioned, you need `mut` here)
```
for field <- fields {
  rec = rec.put(field, data::Value::None)
}
```

Another totally fine option is adding a `put_all` method on `rec` to handle this case